Feature/SOF-6727

VsevolodX opened the pull request:
Yes. NestedDropdown has a lot in common with existing Dropdown, so we need discuss later how to create universal version, so that it's used the same way and looks the same.

unsigned6 committed:
update: add ability to pass color prop in LinearProgress

VsevolodX committed:
=initial draft

VsevolodX committed:
feat: add nested dropdown

VsevolodX committed:
feat: wip1

VsevolodX committed:
feat: add nested dropdown

VsevolodX committed:
feat: add nested dropdown

VsevolodX committed:
update: add JSDoc descripton

VsevolodX committed:
Remove .DS_Store from the repository

VsevolodX committed:
cleanup: remove erroneous files from commit

VsevolodX committed:
fix: rightIcon not showing

VsevolodX committed:
update: add popper auto-positioning and default chevron icons

zoomchik commented on src/mui/components/dropdown/DropdownItem.tsx:
```diff
@@ -4,36 +4,87 @@ import { green } from "@mui/material/colors";
+    const [submenuOpen, setSubmenuOpen] = useState(false);
+    const handleMouseOver = useCallback(() => setSubmenuOpen(true), []);
+    const handleMouseOut = useCallback(() => setSubmenuOpen(false), []);
+    const handleClose = useCallback(() => setSubmenuOpen(false), []);
```
handleOpen and handleClose would be enough instead of handleMouseOver, handleMouseOut .

VsevolodX replied:
Yes. This is an erroneous change, should not have been committed.

zoomchik commented on src/mui/components/nested-dropdown/NestedDropdown.tsx:
```diff
@@ -0,0 +1,196 @@
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [opened, setOpened] = useState(false);
+    const theme = useTheme();
+    const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
```
I think it's better to move mediaQuery to host app. In different places it could require different breakpoint triggers.

timurbazhirov commented on src/mui/components/nested-dropdown/NestedDropdown.tsx:
```diff
@@ -0,0 +1,196 @@
+    contentObject?: React.ReactNode;
+}
+
+// TODO: discuss turning this into fully reusable component, renaming to Dropdown? and changing existing implementations to use that.
```
Create a task for this

VsevolodX committed:
update: remove unused import

VsevolodX committed:
update: remove files from previous commit

VsevolodX commented on src/mui/components/linear-progress/LinearProgress.tsx:
```diff
@@ -17,6 +17,8 @@ export interface ExtendedLinearProgressProps extends LinearProgressProps {
+export default function LinearProgress({ percent, variant, color }: ExtendedLinearProgressProps) {
+    return (
+        <LinearProgressMui variant={variant} value={percent} color={color || getColor(percent)} />
+    );
```
Looks like a pre-commit lint:fix did this. 

Code pulled from main is exactly the same:
![image](https://github.com/Exabyte-io/cove.js/assets/79542055/725eb0bf-90f6-4468-8666-b2fa3af5ccbb)

VsevolodX committed:
feat: add correct display for contentObject

The click on it closes the menu. It's an issue, isn't solved by passing shoudMenuStayOpened = true for some reason.

VsevolodX committed:
fix: menu stays opened when specified

VsevolodX committed:
feat: add ability to style Paper

VsevolodX committed:
update: minor fixes and linting errors fix
